front/src/modules/ui/navbar/components/NavCollapseButton.tsx

```diff
@@ -29,15 +32,27 @@ const StyledCollapseButton = styled.button`
   padding: 0;
 
   user-select: none;
+  visibility: ${({ isHovered }) => (isHovered ? 'visible' : 'hidden')};
   width: 32px;
+
+  @keyframes fadeIn {
+    0% {
+      opacity: 0;
+    }
+    100% {
+      opacity: 1;
+    }
+  }
 `;
 
 type CollapseButtonProps = {
   direction?: 'left' | 'right';
+  isHovered: boolean;
```

Comment:
`isHovered`is a bit misleading for this component as it reflects whether the button should show, not its hover status.
```suggestion
  isVisible: boolean;
```